File: backend/createEmail.py
import os
from dotenv import load_dotenv
import asyncio
from sib_api_v3_sdk import ApiClient, TransactionalEmailsApi
from sib_api_v3_sdk import Configuration
import logging

load_dotenv()

# Configure API key
api_key = os.environ.get("YOUR_API_V3_KEY")
configuration = Configuration()
configuration.api_key['api-key'] = api_key

# Create an API instance
api_instance = TransactionalEmailsApi(ApiClient(configuration))

async def send_transactional_email(subject, sender_name, sender_email, html_content, recipient_email):
    email = {
        "sender": {"name": sender_name, "email": sender_email},
        "to": [{"email": recipient_email}],  # Correct format for Sendinblue API
        "subject": subject,
        "htmlContent": html_content,
    }

    try:
        # Run send_transac_email in a separate thread
        response = await asyncio.to_thread(api_instance.send_transac_email, email)
        print("Email sent successfully:", response)
        return response
    except Exception as error:
        print("Error sending email:", error)
        raise error

File: backend/createEmail.js
const SibApiV3Sdk = require('sib-api-v3-sdk');
const defaultClient = SibApiV3Sdk.ApiClient.instance;

const apiKey = defaultClient.authentications['api-key'];
apiKey.apiKey = process.env.YOUR_API_V3_KEY;


const apiInstance = new SibApiV3Sdk.TransactionalEmailsApi();

async function send_transactional_email(subject, senderName, senderEmail, htmlContent, recipientEmail) {
    const email = {
        sender: { name: senderName, email: senderEmail },
        to: [{ email: recipientEmail }], // Correct format
        subject: subject,
        htmlContent: htmlContent,
    };

    try {
        const response = await apiInstance.sendTransacEmail(email);
        console.log("Email sent successfully:", response);
        return response;
    } catch (error) {
        console.error("Error sending email:", error);
        throw error;
    }
}

module.exports = { send_transactional_email };
